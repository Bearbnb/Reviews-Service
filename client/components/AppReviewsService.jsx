import React from 'react';
import ReactDOM from 'react-dom';
import $ from 'jquery';
import ReviewList from './ReviewList.jsx';
import ReviewSummary from './ReviewSummary.jsx';

import styles from '../../styles/app.css';

class AppReviewsService extends React.Component {
  constructor(props) {
    super(props);
    this.handleSearch = this.handleSearch.bind(this);
    this.resetHomes = this.resetHomes.bind(this);
    this.state = {
      reviews: [],
      house: null,
      host: null,
      length: 0,
      searched: false,
    };
  } 

  componentDidMount() {
    const idPath = window.location.pathname;
    const id = idPath.substring(1, idPath.length - 1);
    this.getHomeReviews(id);
  }

  getHomeReviews(id) {
    this.setState({ house: id });
    $.ajax({
      method: 'GET',
      url: `/reviews/${id}`,
    })
      .done((data) => {
        this.setState({
          reviews: data.reviews,
          length: data.reviews.length,
          host: data.host,
        });
      })
      .fail(() => console.log('didn\'t go through :('));
  }

  resetHomes() {
    this.getHomeReviews(this.state.house);
    this.setState({
      searched: false,
    });
  }

  handleSearch(input) {
    const newReviews = [];
    this.state.reviews.forEach((review) => {
      if (review.review.includes(input)) {
        newReviews.push(review);
      }
    });
    this.setState({
      reviews: newReviews,
      searched: true,
    });
  }

  render() {
    const props = this.state;
    return (
      <div className={styles.App}>
        <ReviewSummary
          onClick={this.handleSearch}
          reviews={props.reviews}
          searched={props.searched}
          resetHomes={this.resetHomes}
          length ={props.length}
        />
        <ReviewList reviews={props.reviews} host={props.host} />
      </div>
    );
  }
}

export default AppReviewsService;
