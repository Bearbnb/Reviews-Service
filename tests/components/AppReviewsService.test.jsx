import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import AppReviewsService from '../../client/components/AppReviewsService';

describe('AppReviewsService Component', () => {
  it('AppReviewsService should render App to the DOM', () => {
    const wrapper = shallow(<AppReviewsService />);
	  expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call componentDidMount once', () => {
    const spy = jest.spyOn(AppReviewsService.prototype, 'componentDidMount');
    const wrapper = mount(<AppReviewsService />);

    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
});
