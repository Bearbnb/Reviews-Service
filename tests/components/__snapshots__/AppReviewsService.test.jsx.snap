// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppReviewsService Component AppReviewsService should render App to the DOM 1`] = `
<div
  className="App"
>
  <ReviewSummary
    length={0}
    onClick={[Function]}
    resetHomes={[Function]}
    reviews={Array []}
    searched={false}
  />
  <ReviewList
    host={null}
    reviews={Array []}
  />
</div>
`;
